File: Génération_Code_Application/Codex5.3_VistaCanvas/backend/app/schemas/review.py
from datetime import datetime

from pydantic import BaseModel, Field


class ReviewCreate(BaseModel):
    rating: int = Field(ge=1, le=5)
    title: str = Field(min_length=1, max_length=200)
    body: str = Field(min_length=1, max_length=5000)
    image_url: str | None = Field(default=None, max_length=500)


class ReviewUpdate(BaseModel):
    rating: int | None = Field(default=None, ge=1, le=5)
    title: str | None = Field(default=None, min_length=1, max_length=200)
    body: str | None = Field(default=None, min_length=1, max_length=5000)
    image_url: str | None = Field(default=None, max_length=500)


class ReviewImageUrlRequest(BaseModel):
    image_url: str = Field(max_length=500)


class ReviewPublic(BaseModel):
    id: int
    product_id: int
    user_id: int
    username: str
    rating: int
    title: str
    body: str
    image_url: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: Génération_Code_Application/Cursor_AuraWear/frontend/src/types/review.ts
export interface ReviewAuthor {
  id: number;
  username: string;
  first_name: string | null;
}

export interface Review {
  id: number;
  product_id: number;
  rating: number;
  title: string | null;
  body: string;
  image_url: string | null;
  author: ReviewAuthor;
  created_at: string;
  updated_at: string;
}

export interface ReviewCreatePayload {
  rating: number;
  title?: string | null;
  body: string;
  image_url?: string | null;
}
